File: campus-trade-user/src/api/order.ts
import request from '@/utils/request'
import type { PageResult, OrderQueryParams } from '@/types'

export interface OrderItemVO {
  id: number
  orderId: number
  goodsId: number
  goodsTitle: string
  goodsImage: string | null
  price: number
}

export interface OrderVO {
  id: number
  orderNo: string
  buyerId: number
  buyerName: string
  sellerId: number
  sellerName: string
  totalAmount: number
  status: string
  remark: string
  deliveryMethod: number | string
  address: string | null
  deliveryAddress: string | null
  payTime: string | null
  shipTime: string | null
  finishTime: string | null
  cancelTime: string | null
  cancelReason: string | null
  createTime: string
  items: OrderItemVO[]
}

export const createOrder = (data: { goodsId: number; remark?: string; deliveryMethod?: string; deliveryAddress?: string }) =>
  request.post<never, OrderVO>('/order', data)

export const rateOrder = (id: number, rating: number, comment?: string) =>
  request.post(`/order/${id}/rate`, null, { params: { rating, comment } })

export const cancelOrder = (id: number, reason?: string) =>
  request.put(`/order/${id}/cancel`, null, { params: { reason } })

export const payOrder = (id: number) => request.put(`/order/${id}/pay`)

export const shipOrder = (id: number) => request.put(`/order/${id}/ship`)

export const finishOrder = (id: number) => request.put(`/order/${id}/finish`)

export const refundOrder = (id: number, reason?: string) =>
  request.put(`/order/${id}/refund`, null, { params: { reason } })

export const approveRefund = (id: number) =>
  request.put(`/order/${id}/approve-refund`)

export const rejectRefund = (id: number, reason?: string) =>
  request.put(`/order/${id}/reject-refund`, null, { params: { reason } })

export const getOrderDetail = (id: number) => request.get<never, OrderVO>(`/order/${id}`)

export const getBuyerOrders = (params: OrderQueryParams) => request.get<never, PageResult<OrderVO>>('/order/buyer', { params })

export const getSellerOrders = (params: OrderQueryParams) => request.get<never, PageResult<OrderVO>>('/order/seller', { params })

File: campus-trade-server/src/main/java/com/campustrade/entity/Order.java
package com.campustrade.entity;

import com.campustrade.base.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@EqualsAndHashCode(callSuper = true)
public class Order extends BaseEntity {

    private String orderNo;
    private Long buyerId;
    private Long sellerId;
    private BigDecimal totalAmount;
    private String status;
    private LocalDateTime payTime;
    private LocalDateTime shipTime;
    private LocalDateTime finishTime;
    private LocalDateTime cancelTime;
    private String cancelReason;
    private String remark;
    private Integer deliveryMethod;
    private String address;
    private String trackingNo;
    private String tradeNo;
    private String preRefundStatus;
    private Long sellerPaymentConfigId;
}